### PR TITLE
Jessica Hebert -C18 Snow Leopards

### PR DESCRIPTION
**Pull request by hebert87**

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,15 +1,126 @@
...
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+  };
+  const availableLetters = [];
+    
+  for (const [letter,frequency] of Object.entries(letterPool)){
+    for (let i = 0;i< frequency;i++){
```
Comment (yangashley) [inline src/adagrams.js:33]: While JS doesn't care about white spaces, we should add spaces after each semi-colon on line 33.

```
for (let i = 0; i< frequency; i++)

File: src/adagrams.js [hunk 2]
```diff
@@ -1,15 +1,126 @@
...
+      availableLetters.push(letter)}   
+  }
+  const hand=[]
+  for (let i = 0; i < 10; i++) {
+    const thisLetter = availableLetters[Math.floor(Math.random() * availableLetters.length)]
+    hand.push(thisLetter)
+    let letterIndex = availableLetters.indexOf(thisLetter)
+    availableLetters.splice(letterIndex, 1)
```
Comment (yangashley) [inline src/adagrams.js:41]: Could you remove the letter by using splice with `thisLetter` instead?

File: src/adagrams.js [hunk 3]
```diff
@@ -1,15 +1,126 @@
...
+}
 
+return hand;
```
Comment (yangashley) [inline src/adagrams.js:44]: This line should be indented once

File: src/adagrams.js [hunk 4]
```diff
@@ -1,15 +1,126 @@
...
+};
+    
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const upperCase = input.toUpperCase();
+
+  for (let letter of upperCase) {
```
Comment (yangashley) [inline src/adagrams.js:50]: Do you ever reassign letter in the for loop? It doesn't look like it so we should use const instead of let

File: src/adagrams.js [hunk 5]
```diff
@@ -1,15 +1,126 @@
...
+  X: 8,
+  Y: 4,
+  Z: 10
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+
+  let score = 0;
+  let bonus = 8;
```
Comment (yangashley) [inline src/adagrams.js:92]: Nice, I like that you put the extra bonus points in a variable. Since it is a constant value that we hardcode to 8 and don't change, we can rename it BONUS. 

Also, since you don't reassign `bonus`, we should use const

File: src/adagrams.js [hunk 6]
```diff
@@ -1,15 +1,126 @@
...
+  word = word.toUpperCase();
+
+  for (let i = 0; i < word.length; i++) {
+    score += letterScores[word[i]];
+  }
```
Comment (yangashley) [inline src/adagrams.js:97]: You could also use a for-of loop here to iterate over ever letter in word.

```
for (letter of word) {
   score += letterScores[letter]
}

File: src/adagrams.js [hunk 7]
```diff
@@ -1,15 +1,126 @@
...
+  let highestScore = 0;
+
+  for (let word of words){
+    let score = scoreWord(word);
+    if (score> highestScore){
+      highestScore = score;
+      highestWord = word;
+    }else if (score == highestScore){
+      if(highestWord.length==10){
+        break
```
Comment (yangashley) [inline src/adagrams.js:118]: Do you want to break out of the loop or continue on with the loop?

File: src/adagrams.js [hunk 8]
```diff
@@ -1,15 +1,126 @@
...
+  return score;
+    
 };
 
 export const highestScoreFrom = (words) => {
   // Implement this method for wave 4
+  let highestWord = '';
+  let highestScore = 0;
+
+  for (let word of words){
```
Comment (yangashley) [inline src/adagrams.js:111]: Since you're not reassigning word we should use const

**Review by yangashley — COMMENTED** [review]
Nice work on js-adagrams. I called out a couple of places regarding adding white spaces -- even though JS doesn't care, adding white spaces where they should be per the JS style guide, helps our eyes read JS code more easily.

There are a few places where you used let when const should be better since the value wasn't being reassigned.  When in doubt, use const and when you try to reassign the variable then Javascript will complain and you can refactor it to let. 

I did notice you didn't update the test file which meant two tests were failing. When you see failing tests, be sure to see why they're failing (does your code handle all the edge cases? or do you have to finish filling in a test?). If you look in Learn, you should see the failed tests. Please go back and fill in the test assertions where you see  "throw "Complete test by adding an assertion"; and push your changes up again. Thanks!


Overall your code was readable and it seems like you've got a good handle on JS so far!
